mypy/plugins/default.py

```diff
@@ -372,6 +372,10 @@ def typed_dict_setdefault_callback(ctx: MethodContext) -> Type:
             )
             return AnyType(TypeOfAny.from_error)
 
+        assigned_readonly_keys = ctx.type.readonly_keys & set(keys)
+        if assigned_readonly_keys:
+            ctx.api.msg.readonly_keys_mutated(assigned_readonly_keys, context=ctx.context)
```

Comment:
Btw, it were covered by 

```python
        assigned_readonly_keys = actual_keys & callee.readonly_keys
        if assigned_readonly_keys:
            self.msg.readonly_keys_mutated(assigned_readonly_keys, context=context)
```

But, since I rmeoved these lines now, `.update` and `|=` requires an extra change.

Reply:
Hm, I now start to think that my change to `check_typeddict_call_with_kwargs` was correct 🤔  
Without it, neither `.update` nor `|=` raise any errors. And it seems like I can't do anything in `typed_dict_update_signature_callback`

Reply:
Ok, this test proves me wrong:

```python
from typing import ReadOnly, TypedDict, NotRequired

class TP(TypedDict):
    key: ReadOnly[str]
    other: NotRequired[ReadOnly[int]]
    mutable: bool

class Mutable(TypedDict):
    mutable: bool

class Regular(TypedDict):
    key: str

tp: TP
m: Mutable
r: Regular

tp1: TP = {**tp, **m}  # E: ReadOnly TypedDict keys ("key", "other") TypedDict are mutated
tp2: TP = {**r, **m}  # E: ReadOnly TypedDict key "key" TypedDict is mutated
```

More work on `.update` and other is needed.